python/sglang/srt/managers/scheduler.py

```diff
@@ -332,6 +332,14 @@ def event_loop_normal(self):
             self.process_input_requests(recv_reqs)
 
             batch = self.get_next_batch_to_run()
+
+            if (
+                self.server_args.enable_dp_mla
+                and self.check_dp_batch_running(batch)
+                and batch is None
+            ):
+                batch = self.get_idle_batch()
```

Comment:
can we move this logic inside `self.get_next_batch_to_run()`?

Reply:
Make sense.

Reply:
The code is updated. There are many return branchs in `get_next_batch_to_run()` and the idle case should be handled based on the returned batch. So I think it's easier to handle it out of `get_next_batch_to_run()`.